src/parser.mly

```diff
@@ -0,0 +1,142 @@
+/* Ocamlyacc parser for SIFT */
+%{ 
+  open Ast 
+%}
+
+/* Arithmetic Operators */
+%token ADD SUBTRACT MULTIPLY DIVIDE MOD
+/* Relational Operators */
+%token EQ LEQ GT GEQ NEQ LT
+/* Logical Operators */
+%token AND OR NOT XOR
+/* Paranthesis and Brackets */
+%token LPAREN RPAREN LBRACE RBRACE LARROW RARROW LSQBRACE RSQBRACE
+/* Delimiters */
+%token COMMA SEMICOLON COLON DOT
+/* Assignment */
+%token ASSIGN 
+/* Features */
+%token PIPE
+/* Keywords */
+%token IF ELIF ELSE SWITCH CASE FOR WHILE CONTINUE BREAK DEFAULT LAMBDA DEFINE EXIT RETURN NULL TRUE FALSE
```

Comment:
I don't think we decided to support `ELIF`

Reply:
Yeah. I wrote it for convenience. Else we need to keep looking after an else whether its an else or else if. For semantic checker this could be more lines of code.